DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces_overview/csr-pipes/Readme.md

```diff
@@ -0,0 +1,190 @@
+# CSR Data
+This implementation uses a register-mapped invocation interface, and demonstrates how to use host pipes to return output into the CSR using the `protocol_avalon_mm` pipe protocol.
+
+![](../assets/csr_out.svg)
+
+## Invocation Interface
+By default, an un-decorated oneAPI kernel will have all its control signals and arguments mapped into the IP component's control/status register (CSR).
+
+## Data Interface - CSR Pipe
+In this design, input `InputPipeA` and `InputPipeB` are implemented as host pipes, as in [Streaming Data](../host-pipes/). However, the output `OutputPipeC` is implemented to write a single vector sum to the IP component's CSR.
+
+To configure a host pipe to map to the CSR, set its protocol as `AVALON_MM` or `AVALON_MM_USES_READY` (from `sycl::ext::intel::prototype::internal` namespace). See how to [specify pipe protocol](../host-pipes/Readme.md#Host-Pipe-Properties).
```

Comment:
i think these protocol names and namespaces have changed. double check to make sure they're up to date.